resources/validations/src/ssp.sch

```diff
@@ -438,43 +409,1475 @@
+                        correct system attribute.</sch:assert>
+            <sch:assert diagnostics="categorization-has-information-type-id-diagnostic"
+                        id="categorization-has-information-type-id"
+                        role="error"
+                        test="oscal:information-type-id">A FedRAMP OSCAL SSP information-type categorization must have at least one
+                        information-type-id.</sch:assert>
+            <!-- FIXME: https://github.com/18F/fedramp-automation/blob/master/documents/Guide_to_OSCAL-based_FedRAMP_System_Security_Plans_(SSP).pdf page 11 has schema error -->
+            <!--        confidentiality-impact, integrity-impact, availability-impact are children of <information-type> -->
+        </sch:rule>
+        <sch:rule context="oscal:information-type-id">
+            <sch:let name="information-types"
+                     value="doc(concat($registry-base-path, '/information-types.xml'))//fedramp:information-type/@id" />
+            <!-- note the variant namespace and associated prefix -->
+            <sch:assert diagnostics="has-allowed-information-type-id-diagnostic"
+                        id="has-allowed-information-type-id"
+                        role="error"
+                        test="current()[. = $information-types]">A FedRAMP OSCAL SSP information-type-id must have a SP 800-60v2r1
+                        identifier.</sch:assert>
+        </sch:rule>
+        <sch:rule context="oscal:confidentiality-impact | oscal:integrity-impact | oscal:availability-impact">
+            <sch:assert diagnostics="cia-impact-has-base-diagnostic"
+                        id="cia-impact-has-base"
+                        role="error"
+                        test="oscal:base">A FedRAMP OSCAL SSP information-type confidentiality-, integrity-, or availability-impact must have a base
+                        element.</sch:assert>
+            <sch:assert diagnostics="cia-impact-has-selected-diagnostic"
+                        id="cia-impact-has-selected"
+                        role="error"
+                        test="oscal:selected">A FedRAMP OSCAL SSP information-type confidentiality-, integrity-, or availability-impact must have a
+                        selected element.</sch:assert>
+        </sch:rule>
+        <sch:rule context="oscal:base | oscal:selected">
+            <sch:let name="fips-levels"
+                     value="('fips-199-low', 'fips-199-moderate', 'fips-199-high')" />
+            <sch:assert diagnostics="cia-impact-has-approved-fips-categorization-diagnostic"
+                        id="cia-impact-has-approved-fips-categorization"
+                        role="error"
+                        test=". = $fips-levels">A FedRAMP OSCAL SSP information-type confidentiality-, integrity-, or availability-impact base or
+                        select element must have an approved value.</sch:assert>
+        </sch:rule>
```

Comment:
Should we not be loading these from the FedRAMP values file?

Reply:
Yes. I may have done this when looking at another L/M/H that was just low|moderate|high with and without initial capitalization. The most recent commit uses the external document.